id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/entity/AnonymousProfileDto.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * 
+ */
+package io.mosip.idrepository.identity.entity;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * @author Neha Farheen
+ *
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class AnonymousProfileDto {
+	
+	private byte[] oldUinData;
+
+	private byte[] newUinData;
+	
+	private String regId;
+
+	private String oldCbeff;
+
+	private String newCbeff;
+	
+	private String uinHash;
+	
+	private String oldCbeffRefId;
+	
+	private String newCbeffRefId;
+	
+	
+	public AnonymousProfileDto setOldUinData(byte[] oldUinData) {
+		this.oldUinData = oldUinData;
+		return this;
+	}
+
+	public AnonymousProfileDto setNewUinData(byte[] newUinData) {
+		this.newUinData = newUinData;
+		return this;
+	}
+
+	public AnonymousProfileDto setOldCbeff(String oldCbeff) {
+		this.oldCbeff = oldCbeff;
+		return this;
+	}
+	
+	public boolean isOldCbeffPresent() {
+		return Objects.nonNull(this.oldCbeff);
+	} 
+
+	public AnonymousProfileDto setNewCbeff(String newCbeff) {
+		this.newCbeff = newCbeff;
+		return this;
+	}
+
+	public boolean isNewCbeffPresent() {
+		return Objects.nonNull(this.newCbeff);
+	}
+
+	public AnonymousProfileDto setOldCbeff(String uinHash, String fileRefId) {
+		if (Objects.isNull(oldCbeff)) {
+			String substringHash = StringUtils.substringAfter(uinHash, "_");
+			this.uinHash = StringUtils.isBlank(substringHash) ? uinHash : substringHash;
+			this.oldCbeffRefId = fileRefId;
+		}
+		return this;
+	}
+
+	public AnonymousProfileDto setNewCbeff(String uinHash, String fileRefId) {
+		if (Objects.isNull(newCbeff)) {
+			String substringHash = StringUtils.substringAfter(uinHash, "_");
+			this.uinHash = StringUtils.isBlank(substringHash) ? uinHash : substringHash;
+			this.newCbeffRefId = fileRefId;
+		}
```

Comment:
Can we have these a logic in the helper class ? That way the dto is clean.

Reply:
Lets keep the DTO class only with @data lombak, and remove all the methods, so the dto is clean. All the work related to conversion lets do it before in the AnonymousProfilehelper class and pass the final field data to the dto object

Reply:
Now with the current changes all the logic is implemented into helper class

Reply:
Resolved